components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/filter/CaptchaFilter.java

```diff
@@ -65,6 +66,7 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 
         try {
 
+            String redirectURL;
```

Comment:
This `redirectURL` url is only used in error handling. lets take out the redirectURL resolution logic to a private method and directly call that in the error flow where we actually need the redirectURL.